### PR TITLE
[[ Bug 21026 ]] Fix handling of binary URL data in HTML5

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

File: engine/src/em-liburl.js [hunk 1]
```diff
@@ -222,23 +222,17 @@ mergeInto(LibraryManager.library, {
...
-            // thus we base64 encode on this side (and decode in libURL)
+			// To return the raw data back to libURL we use a FileReader
+			// to read the data to an ArrayBuffer then construct a
+			// Uint8Array (which will be converted to MCDataRef) from that.
             var xhr = this;
             if (this.responseType == 'blob') {
-                // the FileReader API allows us to convert blobs to base64
-                // the API prefixes base64 data with "data:text/plain;base64,"
-                // make sure we chop this off
                 var reader = new FileReader();
```
Comment (livecodemichael) [inline engine/src/em-liburl.js]: I think it's possible to set the `responseType` of the request to `arraybuffer`, from which we can create a `Uint8Array` directly, removing the need for the `FileReader`. It shouldn't make any functional difference but is probably the slightly neater way to do things.

**Comment by livecodemichael** [comment]
@livecode-vulcan review ok 5f4c9e8

**Comment by livecode-vulcan** [comment]
💙 review by @livecodemichael ok 5f4c9e81b56891b99be30ab264f43dff6a35a924

**Comment by livecode-vulcan** [comment]
😎 test success 5f4c9e81b56891b99be30ab264f43dff6a35a924

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1301)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1317)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1291)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/244)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/242)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1295)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1321)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1301)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1316)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1300)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1310)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1352)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1291)

**Pull request merged**